redis/src/aio/multiplexed_connection.rs

```diff
@@ -51,56 +54,64 @@ impl<O, E> InFlight<O, E> {
 }
 
 // A single message sent through the pipeline
-struct PipelineMessage<S, I, E> {
+struct PipelineMessage<S, E> {
     input: S,
-    output: PipelineOutput<I, E>,
+    output: PipelineOutput<Value, E>,
     response_count: usize,
 }
 
 /// Wrapper around a `Stream + Sink` where each item sent through the `Sink` results in one or more
 /// items being output by the `Stream` (the number is specified at time of sending). With the
 /// interface provided by `Pipeline` an easy interface of request to response, hiding the `Stream`
 /// and `Sink`.
-struct Pipeline<SinkItem, I, E>(mpsc::Sender<PipelineMessage<SinkItem, I, E>>);
+struct Pipeline<SinkItem, E> {
+    sender: mpsc::Sender<PipelineMessage<SinkItem, E>>,
 
-impl<SinkItem, I, E> Clone for Pipeline<SinkItem, I, E> {
+    push_manager: Arc<ArcSwap<PushManager>>,
```

Comment:
why use arcswap instead of a simple mutex?

Reply:
I didn't want to take a lock each time i need to process push message, also arcswap is already used in the crate